client/dashboard/sites/settings-site-visibility/agency-development-site-launch-modal.tsx

```diff
@@ -0,0 +1,76 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Button,
+	Modal,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { launchSiteMutation } from '../../app/queries';
+import type { Site } from '../../data/types';
+
+interface AgencyDevelopmentSiteLaunchModalProps {
+	site: Site;
+	onClose: () => void;
+}
+
+export default function AgencyDevelopmentSiteLaunchModal( {
+	site,
+	onClose,
+}: AgencyDevelopmentSiteLaunchModalProps ) {
+	const queryClient = useQueryClient();
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const [ isLaunching, setIsLaunching ] = useState( false );
+	const mutation = useMutation( {
+		...launchSiteMutation( site.ID ),
+		onSuccess: () => {
+			queryClient.invalidateQueries( { queryKey: [ 'site', site.slug ] } );
+		},
+	} );
+
+	const handleLaunch = () => {
+		setIsLaunching( true );
+		mutation.mutate( undefined, {
+			onSuccess: () => {
+				createSuccessNotice(
+					__( 'Your site has been launched; now you can share it with the world!' ),
+					{
+						type: 'snackbar',
+					}
+				);
+
+				setIsLaunching( false );
+				onClose();
+			},
+			onError: ( error: Error ) => {
+				createErrorNotice( error.message || __( 'Failed to launch site' ), {
+					type: 'snackbar',
+				} );
+
+				setIsLaunching( false );
+				onClose();
```

Comment:
The `mutate` function also accepts an `onSettled` callback where you can put this sort of code that you want to run regardless of success or error.